src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,594 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.text;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serial;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+import sun.util.locale.provider.LocaleProviderAdapter;
+
+/**
+ * {@code ListFormat} produces or parses a list of concatenated
+ * strings in a locale-sensitive way. Use {@code ListFormat} to construct a list
+ * of strings displayed for end users. For example, displaying 3 weekdays
+ * as "Monday, Wednesday, and Friday". This class provides the functionality
+ * defined in Unicode Consortium's LDML specification for
+ * <a href="https://www.unicode.org/reports/tr35/tr35-general.html#ListPatterns">
+ * List Patterns</a>.
+ * <p>
+ * Three types of concatenation are provided: {@link Type#STANDARD STANDARD},
+ * {@link Type#OR OR}, and {@link Type#UNIT UNIT}, also three styles for each
+ * type are provided: {@link Style#FULL FULL}, {@link Style#SHORT SHORT}, and
+ * {@link Style#NARROW NARROW}. The following snippet is an example of formatting
+ * the list of Strings {@code "Foo", "Bar", "Baz"} in US English with
+ * {@code STANDARD} type and {@code FULL} style:
+ * {@snippet lang=java :
+ * ListFormat.getInstance(Locale.US, ListFormat.Type.STANDARD, ListFormat.Style.FULL)
+ *     .format(List.of("Foo", "Bar", "Baz"))
+ * }
+ * This will produce the concatenated list string, "Foo, Bar, and Baz" as seen in
+ * the following table:
+ * <table class="striped">
+ * <caption style="display:none">Formatting examples</caption>
+ * <thead>
+ * <tr><th scope="col"></th>
+ *     <th scope="col">FULL</th>
+ *     <th scope="col">SHORT</th>
+ *     <th scope="col">NARROW</th></tr>
+ * </thead>
+ * <tbody>
+ * <tr><th scope="row" style="text-align:left">STANDARD</th>
+ *     <td>Foo, Bar, and Baz</td>
+ *     <td>Foo, Bar, &amp; Baz</td>
+ *     <td>Foo, Bar, Baz</td>
+ * <tr><th scope="row" style="text-align:left">OR</th>
+ *     <td>Foo, Bar, or Baz</td>
+ *     <td>Foo, Bar, or Baz</td>
+ *     <td>Foo, Bar, or Baz</td>
+ * <tr><th scope="row" style="text-align:left">UNIT</th>
+ *     <td>Foo, Bar, Baz</td>
+ *     <td>Foo, Bar, Baz</td>
+ *     <td>Foo Bar Baz</td>
+ * </tbody>
+ * </table>
+ * Note that the above chart represents the examples in the CLDR definition,
+ * and there could be different patterns from other locale providers.
+ * <p>
+ * Alternatively, Locale, Type, and/or Style independent instances
+ * can be created with {@link #getInstance(String[])}. The String array to the
+ * method specifies the delimiting patterns for the start/middle/end portion of
+ * the formatted string, as well as optional specialized patterns for two or three
+ * elements. Refer to the method description for more detail.
```

Comment:
```suggestion
 * Alternatively, more flexible patterns can be constructed from the pattern parts for the start, middle, 
 * and end of the formatted list as well as specialized patterns for lists of two or three elements.
 * Refer to the {@link #getInstance(String[])} method description for more detail.
```

Reply:
For this part, I would not use `patterns` in the first suggested sentence but use `more flexible ListFormat instances` instead.